packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/Carousel/useCarouselWalker.ts

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { CAROUSEL_ACTIVE_ITM, CAROUSEL_ITEM } from './constants';
+import { createContext } from 'react';
+import { Context, ContextSelector } from '@fluentui/react-context-selector';
+import { useContextSelector } from '@fluentui/react-context-selector';
+import { isHTMLElement } from '@fluentui/react-utilities';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+
+export const useCarouselWalker_unstable = () => {
+  const { targetDocument } = useFluent();
+
+  const treeWalkerRef = React.useRef<TreeWalker | undefined>(targetDocument?.createTreeWalker(targetDocument.body));
+  const htmlRef = React.useRef<HTMLDivElement | undefined>(targetDocument?.createElement('div'));
```

Comment:
```suggestion
  const htmlRef = React.useRef<HTMLDivElement | null>(null);
```

You'll create a new div on every re-render, I suggest to just use null